Source/WebCore/style/ScopedName.h

```diff
@@ -36,14 +36,29 @@ struct ScopedName {
     ScopeOrdinal scopeOrdinal { ScopeOrdinal::Element };
     bool isIdentifier { true };
 
+    template<typename... F> decltype(auto) switchOn(F&&... f) const
+    {
+        auto visitor = WTF::makeVisitor(std::forward<F>(f)...);
+        if (isIdentifier)
+            return visitor(CustomIdentifier { name });
+        return visitor(name);
+    }
+
     bool operator==(const ScopedName&) const = default;
 };
 
-inline WTF::TextStream& operator<<(WTF::TextStream& ts, ScopedName scopedName)
-{
-    ts << scopedName.name;
-    return ts;
-}
+// MARK: - Conversion
+
+template<> struct CSSValueConversion<ScopedName> {
+    ScopedName operator()(BuilderState&, const CSSPrimitiveValue&);
+    ScopedName operator()(BuilderState&, const CSSValue&);
+};
+
+// MARK: - Logging
+
+WTF::TextStream& operator<<(WTF::TextStream&, const ScopedName&);
 
 } // namespace Style
 } // namespace WebCore
+
+template<> inline constexpr auto WebCore::TreatAsVariantLike<WebCore::Style::ScopedName> = true;
```

Comment:
Why isn't this inside the WebCore namespace?

Reply:
I also think that `constexpr` in cases like this is equivalent to `inline constexpr`.

Reply:
For constants like this, `inline constexpr` is necessary, otherwise it can lead to ODR violations. (I used to have a good blog post to link about this, but I can't find it right now. https://quuxplusone.github.io/blog/2022/07/08/inline-constexpr/ does an ok job explaining it).